### PR TITLE
#164796894 implement user comment on article

### PR DESCRIPTION
**Pull request by Oluwaseyi000**
#### Description
Add REST API endpoint that enables users to comment on  articles

#### Type of change
New feature(a non-breaking change that adds a functionality)

#### How Has This Been Tested?
integration and unit testing

#### Checklist:
My code follows the style guidelines of this project

#### PT
[#164796894](https://www.pivotaltracker.com/story/show/164796894)

### CODE REVIEW

File: src/routers/index.js [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
+import articleController from '../controllers/article';
 
 const router = Router();
 
+router.post('/article/comment', articleController.comment);
+
 router.get('/', (req, res) => res.status(200).json({
   message: 'Welcome to the Authors Haven API',
-}));
+}),);
```
[the remark this answers is not in the document]
Comment (toluola) [reply src/routers/index.js]: This will be fixed soon

**Comment by ssanusi** [comment]
@Oluwaseyi000 kindly resolve merge conflict

File: src/routers/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Oluwaseyi000) [reply src/routers/index.js]: resolved

[the remark this answers is not in the document]
Comment (Oluwaseyi000) [reply src/routers/index.js]: resolved

File: tests/integration/comment.test.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import { createComment, wrongComment } from '../mockup/comment';
+
+const { assert } = chai;
+chai.use(chaiHttp);
+
+describe('user comment of articles', () => {
+  let app = null;
+  let agent = null;
+
+  beforeEach(async () => {
```
[the remark this answers is not in the document]
Comment (Oluwaseyi000) [reply tests/integration/comment.test.js:16]: resolved

File: src/controllers/authentication/twitter.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+const twitterStrategy = new TwitterStrategy(
+  {
+    consumerKey: process.env.consumerKey,
+    consumerSecret: process.env.consumerSecret,
+    callbackURL: 'http://localhost:4000/api/v1/auth/twitter/callback',
+    includeEmail: true,
+  },
+  (token, tokenSecret, profile, done) => done(null, profile),
+);
```
[the remark this answers is not in the document]
Comment (Oluwaseyi000) [reply src/controllers/authentication/twitter.js]: resolved

File: src/controllers/article.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+ */
+class article {
+  /**
+   * @method comment
+   * @description enable authenticated users to comment on existing article
+   * @param {object} req - The Request Object
+   * @param {object} res - The Response Object
+   * @returns {object} - JSON new comment response object
+   */
+  static async comment(req, res) {
```
[the remark this answers is not in the document]
Comment (Oluwaseyi000) [reply src/controllers/article.js]: resolved

**Review by Temmyogunbo — APPROVED** [review]

**Pull request merged**